### PR TITLE
News api 6

### PR DESCRIPTION
**Pull request by rosie562**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -86,6 +86,46 @@ describe("GET /api/articles/:article_id", () => {
...
 
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: an array of comments for the given article_id with the correct properties", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeSortedBy("created_at", { descending: true });
```
Comment (nickkagame) [inline __tests__/app.test.js:97]: please add a length check as this will pass for an empty array, when in some cases it shouldn't!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -86,6 +86,46 @@ describe("GET /api/articles/:article_id", () => {
...
+        comments.forEach((comments) => {
+          expect(comments).toHaveProperty("comment_id");
+          expect(comments).toHaveProperty("votes");
+          expect(comments).toHaveProperty("created_at");
+          expect(comments).toHaveProperty("author");
+          expect(comments).toHaveProperty("body");
+          expect(comments).toHaveProperty("article_id");
+        });
+      });
+  });
```
Comment (nickkagame) [inline __tests__/app.test.js:107]: we'll need another test for Status 200, valid ID, but has no comments responds with an empty array of comments

File: controllers/news.controllers.js [hunk 3]
```diff
@@ -30,8 +32,18 @@ exports.getArticleById = (req, res, next) => {
     .catch(next);
 };
 
+
+exports.getCommentByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+  selectCommentByArticleId(article_id)
```
Comment (nickkagame) [inline controllers/news.controllers.js]: we need to check it the article exists as well.  As this will determine whether no articles being return indicates an article that has no comments, or an article that doesn't exist!

**Review by nickkagame — CHANGES_REQUESTED** [review]
Nice work rosie! There's just a few extra things we need.  See my comments for details   
 (Think back to the lecture in the morning re. a possible solution!)

**Review by nickkagame — COMMENTED** [review]
great work rosie!
